executor/src/witgen/global_constraints.rs

```diff
@@ -236,7 +237,9 @@ fn propagate_constraints<T: FieldElement>(
             }
         }
         Identity::Lookup(LookupIdentity { left, right, .. })
-        | Identity::Permutation(PermutationIdentity { left, right, .. }) => {
+        | Identity::PhantomLookup(PhantomLookupIdentity { left, right, .. })
+        | Identity::Permutation(PermutationIdentity { left, right, .. })
+        | Identity::PhantomPermutation(PhantomPermutationIdentity { left, right, .. }) => {
```

Comment:
I think this is wrong (it was before), if it was a permutation it would be more than just a range constraint, right?

Reply:
Do you mean that we could learn more in the permutation case?

Reply:
I mean that a constraint like [a] is [BYTES] not only enforces a range constraint, but also that each byte appears exactly once. But this function should detect identities that only enforce range constraints, so I think we should only pay attention to lookups here.

Reply:
dOnE